src/components/Button/Button.tsx

```diff
@@ -2,33 +2,77 @@ import React from "react";
 import styled, { css } from "styled-components";
 
 export interface ButtonProps {
-  primary?: boolean;
   children: string;
+  variant: "filled" | "outlined";
+  theme: "light" | "dark" | "blue";
+  disabled: boolean;
   onClick?: () => void;
 }
 
 /**
  * Primary UI component for user interaction
  */
 export const Button = ({ children, ...props }: ButtonProps) => (
-  <StyledButton type="button" {...props}>
-    {children}
-  </StyledButton>
+  <StyledButton {...props}>{children}</StyledButton>
```

Comment:
It could be could to add a wrapper around `onClick` with something like this
```suggestion
  <StyledButton onClick={onSafeClick} {...props}>{children}</StyledButton>
```

```js
function onSafeClick(e) {
  if (!props.disabled)
    props.onClick(e);
}
```
What do you think ?

Reply:
Added with https://github.com/Major-Eth/yearn-web-lib/pull/9/commits/c0ae006fb15957273a616116bde068a3f235dbdd

I'm still unsure if this is needed though as when the button is in the `disabled` state we can't click on it. Easy to bypass though, so I think it doesn't hurt.